### PR TITLE
Cleanup in ngraph_test_utils.hpp/cpp

### PR DESCRIPTION
**Pull request by pelszkow**
Before starting work on `compare_function` I want to do cleanup without changes in logic.

### CODE REVIEW

File: inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.cpp [hunk 1]
```diff
@@ -1,33 +1,39 @@
...
-    const std::string info2Name = isTypeRelaxed(info2.name) && (info2.parent != nullptr) ? info2.parent->name : info2.name;
+    const std::string info1Name =
+        isTypeRelaxed(info1.name) && (info1.parent != nullptr) ? info1.parent->name : info1.name;
+    const std::string info2Name =
+        isTypeRelaxed(info2.name) && (info2.parent != nullptr) ? info2.parent->name : info2.name;
     return info1Name == info2Name;
 }
 
-bool compare_rt_keys(const std::shared_ptr<ngraph::Node>& node1, const std::shared_ptr<ngraph::Node>& node2) {
+template <typename Node>
```
Comment (ilyachur) [inline inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.cpp:35]: Why did you add a template argument for this function? I don't see that we use it for different types.

Comment (pelszkow) [reply inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.cpp:35]: I add it because I thought I replace queue of `shared_ptr` to raw ptr inside `compare_function`. This change should not bring any change after compilation. It's hidden in unnamed namespace and most probably it will be inlined.

**Review by ilyachur — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

**Review by iefode — APPROVED** [review]

**Comment by GlebKazantaev** [comment]
Thanks for merging this PR without my approval...

File: inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.cpp [hunk 2]
```diff
@@ -52,6 +58,44 @@ bool compare_rt_keys(const std::shared_ptr<ngraph::Node>& node1, const std::shar
...
+std::pair<bool, std::string> error(std::string s) {
+    return {false, std::move(s)};
+}
+
+std::string typeInfoToStr(const ngraph::Node::type_info_t& typeInfo) {
+    return std::string(typeInfo.name) + "/" + to_str(typeInfo.version);
+}
+
+template <typename Node>
+std::string name(const Node& n) {
```
Comment (GlebKazantaev) [inline inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.cpp:81]: Why do we need it? I think function that call another functions are useless especially in this case.

File: inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.cpp [hunk 3]
```diff
@@ -70,36 +114,37 @@ std::pair<bool, std::string> compare_functions(
...
 
-    auto typeInfoToStr = [](const ngraph::Node::type_info_t & typeInfo) {
-        return std::string(typeInfo.name) + "/" + std::to_string(typeInfo.version);
-    };
-
     std::ostringstream err_log;
 
-    std::queue<std::pair<std::shared_ptr<ngraph::Node>, std::shared_ptr<ngraph::Node>>> q;
+    using ComparedNodes = std::pair<std::shared_ptr<ngraph::Node>, std::shared_ptr<ngraph::Node>>;
+    std::queue<ComparedNodes> q;
```
Comment (GlebKazantaev) [inline inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.cpp:136]: What is the purpose of ComparedNodes if it is used only in one place?

Comment (pelszkow) [reply inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.cpp:136]: To get meaningful name for generic type like `std::pair`.

File: inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.cpp (hunk 2 above)
Comment (pelszkow) [reply inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.cpp:81]: From my perspective this function reduce boilerplate code.
